lib/ansible/galaxy/api.py

```diff
@@ -133,6 +133,15 @@ def wrapped(self, *args, **kwargs):
                                    % (method.__name__, ", ".join(versions), ", ".join(available_versions),
                                       self.name, self.api_server))
 
+            # Warn only when we know we are talking to a collections API
+            if 'v1' not in versions and {'v3'}.difference(available_versions) == {'v3'}:
```

Comment:
```suggestion
            if common_versions == {'v2'}:
```
Wouldn't this be easier to read and also not trip a false error message on a hypothetical `v4` only server?

Reply:
hrm, seems like it would be functional, just need to think it through again.